### PR TITLE
Update libsecp256k1 subtree to current master

### PR DESCRIPTION
**Pull request by fanquake**
The motivation for this bump is some small build cleanups, including [dropping the `--enable-experimental`](https://github.com/bitcoin-core/secp256k1/commit/80cf4eea5fa0162350614c08f2252a07f9d7804b) flag from the libsecp configure  invocation, as well as some [now-redundant](https://github.com/bitcoin-core/secp256k1/pull/1090) `pkg-config` variable exporting from our own configure. We also get the benefit of a slightly more efficient libsecp configure due to https://github.com/bitcoin-core/secp256k1/pull/1088.

This also includes a change in our code to migrate from using the [now deprecated](https://github.com/bitcoin-core/secp256k1/commit/99e6568fc6ea2768f5355eb4617283086f756931) `secp256k1_schnorrsig_sign` to `secp256k1_schnorrsig_sign32`.

Guix Build (on x86_64):
```bash
b9f6ad90c75f7edd7c4444c6c3401d8b6ab29a8da22ae22ddaedd94688227b5d  guix-build-404c53062bb8/output/aarch64-linux-gnu/SHA256SUMS.part
250d47ae299d8385d5590518fa2adaabde76e2566fd27e12bf36b62663d13e13  guix-build-404c53062bb8/output/aarch64-linux-gnu/bitcoin-404c53062bb8-aarch64-linux-gnu-debug.tar.gz
48d610dc6f5169f925f782571dac2f082695f89008beadad4adef4c1b583a612  guix-build-404c53062bb8/output/aarch64-linux-gnu/bitcoin-404c53062bb8-aarch64-linux-gnu.tar.gz
8f04ee26e4079719e3935bd0e4287cc11a2a16875bf01e2a63d67492a1fa5367  guix-build-404c53062bb8/output/arm-linux-gnueabihf/SHA256SUMS.part
7d7d7fcfb032bda92e53abd8d608257f0ef17b1e3e52a1414260b896786fb2dc  guix-build-404c53062bb8/output/arm-linux-gnueabihf/bitcoin-404c53062bb8-arm-linux-gnueabihf-debug.tar.gz
30bae2ff3d044f4e39f992a68f6b296b7be2aea350bca4a0415c739a32c20bd9  guix-build-404c53062bb8/output/arm-linux-gnueabihf/bitcoin-404c53062bb8-arm-linux-gnueabihf.tar.gz
5f550fb0b950250eeffce3480ec6403530b0880570a5860ef6c32a3e92eac92f  guix-build-404c53062bb8/output/arm64-apple-darwin/SHA256SUMS.part
c10664d13aeec8c860bf72be833c738973ae18e4d28cdf08b2f9bee960ebff1d  guix-build-404c53062bb8/output/arm64-apple-darwin/bitcoin-404c53062bb8-arm64-apple-darwin-unsigned.dmg
becab75b11cf4ca6f559f8eef835f3574629f6eb932ac716ed4f8c044a85831f  guix-build-404c53062bb8/output/arm64-apple-darwin/bitcoin-404c53062bb8-arm64-apple-darwin-unsigned.tar.gz
bc86433652fe3552f6a13088191364ae7514c9fe3a244da86a6db096bb4922fc  guix-build-404c53062bb8/output/arm64-apple-darwin/bitcoin-404c53062bb8-arm64-apple-darwin.tar.gz
1f585cb9a1356343df4b2726ecfe2598c9903304afb047c047c2cef318555dd3  guix-build-404c53062bb8/output/dist-archive/bitcoin-404c53062bb8.tar.gz
9ede534ba2c6cecb550473eead195627327e826ebb0118e23d60ab482d40e241  guix-build-404c53062bb8/output/powerpc64-linux-gnu/SHA256SUMS.part
77ddb7d7d639b1dd4508468a8ef27e45b35c8b2f8624584a70e6b64798a4ea7a  guix-build-404c53062bb8/output/powerpc64-linux-gnu/bitcoin-404c53062bb8-powerpc64-linux-gnu-debug.tar.gz
36178c1f1c12942ff05275daa3570f8b45419ee8d9f391d750afb405219986f0  guix-build-404c53062bb8/output/powerpc64-linux-gnu/bitcoin-404c53062bb8-powerpc64-linux-gnu.tar.gz
8a15a4da7a9a5e00c49d9aeedf3c6fc666c0d230be1369eac7caf4571d5905e0  guix-build-404c53062bb8/output/powerpc64le-linux-gnu/SHA256SUMS.part
400c58113f2d07c87e03c8528b292c6aca808a2bccae4b041cad3a26a05b6aad  guix-build-404c53062bb8/output/powerpc64le-linux-gnu/bitcoin-404c53062bb8-powerpc64le-linux-gnu-debug.tar.gz
3b9f9d8614ac3a27416e53354b2b0a64d364f91493e9d0f41583a6f492546824  guix-build-404c53062bb8/output/powerpc64le-linux-gnu/bitcoin-404c53062bb8-powerpc64le-linux-gnu.tar.gz
98506b23ee08ad8af958f816da2e4518d661e88d5c6308de1f5e3b2fc787b86c  guix-build-404c53062bb8/output/riscv64-linux-gnu/SHA256SUMS.part
c701a7b77cea4fdc2588b511f1b2c71b89c83bfba19fdb2ac113a5a4b14ac392  guix-build-404c53062bb8/output/riscv64-linux-gnu/bitcoin-404c53062bb8-riscv64-linux-gnu-debug.tar.gz
34d58e6392cd58b3c76e30cd8600c0dbefba7e9c6d5df78c3ef23e81c4e4d26a  guix-build-404c53062bb8/output/riscv64-linux-gnu/bitcoin-404c53062bb8-riscv64-linux-gnu.tar.gz
92fa30e9c6d81dd1e1514b65d3e1abe68ded897237cd99f66aa760d445109c04  guix-build-404c53062bb8/output/x86_64-apple-darwin/SHA256SUMS.part
bee180b02f178ae9980ef159f65913a71cbd037c4aff5f2906af5f174a677da3  guix-build-404c53062bb8/output/x86_64-apple-darwin/bitcoin-404c53062bb8-x86_64-apple-darwin-unsigned.dmg
ad7d18d779ab7a7944817d1f368d0a6bdd174bf1211b0f90180c8ccf04ec4062  guix-build-404c53062bb8/output/x86_64-apple-darwin/bitcoin-404c53062bb8-x86_64-apple-darwin-unsigned.tar.gz
7489d1d5d48ad95cf58bb11b5fdeccadac6fa758784fb498529fca2330abe069  guix-build-404c53062bb8/output/x86_64-apple-darwin/bitcoin-404c53062bb8-x86_64-apple-darwin.tar.gz
74660fb0ebce2a08b03980a57bffcad62e078dc967a74d2395660ff51c019640  guix-build-404c53062bb8/output/x86_64-linux-gnu/SHA256SUMS.part
cd377fa6b46276c2f8a32e199e6f9adf6aa67315688656709d6dc0744d54a837  guix-build-404c53062bb8/output/x86_64-linux-gnu/bitcoin-404c53062bb8-x86_64-linux-gnu-debug.tar.gz
919c521950369d8ad46db2d15b00abb488abfb080d157a41b2db429122a428ed  guix-build-404c53062bb8/output/x86_64-linux-gnu/bitcoin-404c53062bb8-x86_64-linux-gnu.tar.gz
2debca995d432965a8786b6ff74aed42e9e2f1cb0fecbe2d9fc5b850c192fcff  guix-build-404c53062bb8/output/x86_64-w64-mingw32/SHA256SUMS.part
e33169f684fb031ec18ed39812617d3eb263257f6c7564b8f4c974ad05fe672c  guix-build-404c53062bb8/output/x86_64-w64-mingw32/bitcoin-404c53062bb8-win64-debug.zip
029d0a4180cb908d517fcf689dcf46d42fbf383e11dc609711617066ae039ab0  guix-build-404c53062bb8/output/x86_64-w64-mingw32/bitcoin-404c53062bb8-win64-setup-unsigned.exe
7e349c688cac66436562c4805f420b0536db5a3b3abf54d0e8c7752f59874a5c  guix-build-404c53062bb8/output/x86_64-w64-mingw32/bitcoin-404c53062bb8-win64-unsigned.tar.gz
1bff98e82e95c93d6060227408502f5e2d8597d526b912cb6dc0a90ae3094a8f  guix-build-404c53062bb8/output/x86_64-w64-mingw32/bitcoin-404c53062bb8-win64.zip
```

### CODE REVIEW

**Comment by fanquake** [comment]
@sipsorcery could you take a look at the MSVC failure?

**Comment by fanquake** [comment]
Guix build (on x86_64):
```bash
5a81a5da7e3e8c8bf5d79bedd42d028c6f9954a610b01c30dd4a4d14d1c1a984  guix-build-0936101ab45a/output/aarch64-linux-gnu/SHA256SUMS.part
8e58e5656dedd5e2376f17230cb20f17452197ae886711728eaf35fdcf29465f  guix-build-0936101ab45a/output/aarch64-linux-gnu/bitcoin-0936101ab45a-aarch64-linux-gnu-debug.tar.gz
35c2fc1d50ebe2e9e87612c668faf270d68733397a4eb7d242a6463a5802e996  guix-build-0936101ab45a/output/aarch64-linux-gnu/bitcoin-0936101ab45a-aarch64-linux-gnu.tar.gz
4d2bb757342d9c16bc72b3391944058d7be2b0aa7630c6dccaa65605cd55f1c4  guix-build-0936101ab45a/output/arm-linux-gnueabihf/SHA256SUMS.part
6857434e1a1740e4f180a4f32c58cb1ebe48c401a16eab241b06a3e1fce5b9cc  guix-build-0936101ab45a/output/arm-linux-gnueabihf/bitcoin-0936101ab45a-arm-linux-gnueabihf-debug.tar.gz
12e233ca0de582f900ae32684504f2ca54c5891ef2e3ebaa4d127f69e059e48b  guix-build-0936101ab45a/output/arm-linux-gnueabihf/bitcoin-0936101ab45a-arm-linux-gnueabihf.tar.gz
fe02026316bc97303b22df5c48580b26023f4f87b19974e02f6ef6d4c0ccd95f  guix-build-0936101ab45a/output/arm64-apple-darwin/SHA256SUMS.part
d5e1a4be0c27900f77d13a8ae5842531486412ef58d22d17422d5dc40630236c  guix-build-0936101ab45a/output/arm64-apple-darwin/bitcoin-0936101ab45a-arm64-apple-darwin-unsigned.dmg
ebc5398b5c0e31e7d76a49ff5ddb176aadf48e695fb9f05e5a9ebf862939cdb5  guix-build-0936101ab45a/output/arm64-apple-darwin/bitcoin-0936101ab45a-arm64-apple-darwin-unsigned.tar.gz
7f1aba48c11972bb27e2cae807339def5625118e9d81ee6284e0054de1fac0a2  guix-build-0936101ab45a/output/arm64-apple-darwin/bitcoin-0936101ab45a-arm64-apple-darwin.tar.gz
7242942a00c035b5c845ae0150e7dd8ca071ebef3088d5088c42b8cabf9cd011  guix-build-0936101ab45a/output/dist-archive/bitcoin-0936101ab45a.tar.gz
895089abbf6b93c0d49b380d2325e40c7e1e6d312deca9b28b26bf49889d404c  guix-build-0936101ab45a/output/powerpc64-linux-gnu/SHA256SUMS.part
1b2db96224e107765490e9e631bf25adef7960cdafa1bc0a1cad1965b6a511c8  guix-build-0936101ab45a/output/powerpc64-linux-gnu/bitcoin-0936101ab45a-powerpc64-linux-gnu-debug.tar.gz
5d3ca255eea120e4636a29ea43e726ccb6a1861c82f4f99e1d0d96d67dddb1be  guix-build-0936101ab45a/output/powerpc64-linux-gnu/bitcoin-0936101ab45a-powerpc64-linux-gnu.tar.gz
031709cf360aa0d7e91a80a10cd98f109956f03dc71f098e8626331a3a9a51f5  guix-build-0936101ab45a/output/powerpc64le-linux-gnu/SHA256SUMS.part
9ff1b4b7abe5d7851f116ed547e291b42d93565ddd20c98533e9cf1085f0bd81  guix-build-0936101ab45a/output/powerpc64le-linux-gnu/bitcoin-0936101ab45a-powerpc64le-linux-gnu-debug.tar.gz
1b87d79ad380b11cc8d3c669feac81f3975d46953a32ed7b37712da2d7107f3f  guix-build-0936101ab45a/output/powerpc64le-linux-gnu/bitcoin-0936101ab45a-powerpc64le-linux-gnu.tar.gz
e6cadb4e03251104a0a60fd17e09ff630f71dbf6393a9487d027664dc7234e9e  guix-build-0936101ab45a/output/riscv64-linux-gnu/SHA256SUMS.part
3570b071442587d329da40f20b5a44325a73962f21c1e0436cc48090119100a7  guix-build-0936101ab45a/output/riscv64-linux-gnu/bitcoin-0936101ab45a-riscv64-linux-gnu-debug.tar.gz
131131e470e2a6dad83d3b0d7f3076a971521f26640b8c7eea4f6ae1298f09b4  guix-build-0936101ab45a/output/riscv64-linux-gnu/bitcoin-0936101ab45a-riscv64-linux-gnu.tar.gz
41ca7060caf8aef56591e3013143fef3ab385d61c417b619d8e2246c5238378e  guix-build-0936101ab45a/output/x86_64-apple-darwin/SHA256SUMS.part
ab164b25c8f105ea4503ed99a64aabc8a70eb923f24f07d397148316f87e5db5  guix-build-0936101ab45a/output/x86_64-apple-darwin/bitcoin-0936101ab45a-x86_64-apple-darwin-unsigned.dmg
ae8082102bc48a75c03472893b1ff5950d74c53ef37940cd5ed30befab7effdb  guix-build-0936101ab45a/output/x86_64-apple-darwin/bitcoin-0936101ab45a-x86_64-apple-darwin-unsigned.tar.gz
cd0cda5879db4ce47132b263b0ec905d70988c17521c39b12c9cf87cccc97c9d  guix-build-0936101ab45a/output/x86_64-apple-darwin/bitcoin-0936101ab45a-x86_64-apple-darwin.tar.gz
3f80b92ccea6c1a07165e85e88b2e96c99797a641cf38d5c0e6438ac2eb73c14  guix-build-0936101ab45a/output/x86_64-linux-gnu/SHA256SUMS.part
c10d9eb0d6be2a8f5d56fdc1c7666fb37d77847300680763a2336811f2800d39  guix-build-0936101ab45a/output/x86_64-linux-gnu/bitcoin-0936101ab45a-x86_64-linux-gnu-debug.tar.gz
ab3a7b43f1362cfbeb6715cbbbeaa0342f68c813a816804e6aaedaf5d7ba3290  guix-build-0936101ab45a/output/x86_64-linux-gnu/bitcoin-0936101ab45a-x86_64-linux-gnu.tar.gz
2b73a505cbc604381c288d61da5f120594329c1b3c2fb4809db25349c1292071  guix-build-0936101ab45a/output/x86_64-w64-mingw32/SHA256SUMS.part
dd457c1ff9518e208f2cc38bff22e01277fd27324b36ba3ba61a2d8cad9d48f0  guix-build-0936101ab45a/output/x86_64-w64-mingw32/bitcoin-0936101ab45a-win64-debug.zip
196896274a9dd9c62ebaca47016a76ded7ef8ff09900ac26705633732a213233  guix-build-0936101ab45a/output/x86_64-w64-mingw32/bitcoin-0936101ab45a-win64-setup-unsigned.exe
c7157cefe674b2787ff63600754737925356b35897d00f2a807c024c9f9b72da  guix-build-0936101ab45a/output/x86_64-w64-mingw32/bitcoin-0936101ab45a-win64-unsigned.tar.gz
16cec24555c868d0361420ab296b80341cb034e5fca4b0f245ca79ded75ae0a7  guix-build-0936101ab45a/output/x86_64-w64-mingw32/bitcoin-0936101ab45a-win64.zip
```

**Comment by sipsorcery** [comment]
@fanquake I think the msvc failure is the same problem that was fixed on #23432.

**Comment by fanquake** [comment]
> @fanquake I think the msvc failure is the same problem that was fixed on https://github.com/bitcoin/bitcoin/pull/23432.

Thanks. Have pulled that commit in here.

File: src/key.cpp [hunk 1]
```diff
@@ -288,7 +288,7 @@ bool CKey::SignSchnorr(const uint256& hash, Span<unsigned char> sig, const uint2
         uint256 tweak = XOnlyPubKey(pubkey_bytes).ComputeTapTweakHash(merkle_root->IsNull() ? nullptr : merkle_root);
         if (!secp256k1_keypair_xonly_tweak_add(GetVerifyContext(), &keypair, tweak.data())) return false;
     }
-    bool ret = secp256k1_schnorrsig_sign(secp256k1_context_sign, sig.data(), hash.data(), &keypair, aux.data());
+    bool ret = secp256k1_schnorrsig_sign32(secp256k1_context_sign, sig.data(), hash.data(), &keypair, aux.data());
```
Comment (laanwj) [inline src/key.cpp:291]: The PR for this function renaming (with further rationale) is: bitcoin-core/secp256k1#1089

**Comment by laanwj** [comment]
Code review ACK 404c53062bb80853d5967187bdb7b5f7e749de7f, I checked the changes to our tree thoroughly but didn't review all upstream secp256k1 changes in detail.

Subtree check was successful:
```
$ test/lint/git-subtree-check.sh src/secp256k1
src/secp256k1 in HEAD currently refers to tree 1df8d9a97705fe9a2a3e0bbeb030e9de22739633
src/secp256k1 in HEAD was last updated in commit afb7a6fe06a33956ef43429d31f5934448f6e671 (tree 1df8d9a97705fe9a2a3e0bbeb030e9de22739633)
GOOD
```

**Comment by fanquake** [comment]
cc @real-or-random @jonasnick

**Review by real-or-random — APPROVED** [review]
utACK 404c53062bb80853d5967187bdb7b5f7e749de7f I reviewed the diff to Core, I'm with updating to libsecp256k1 master, but I haven't verified that the libsecp256k1 tree here has been updated correctly

**Comment by gruve-p** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/24792/commits/404c53062bb80853d5967187bdb7b5f7e749de7f

**Comment by dunxen** [comment]
Hmm. Getting a configure error on this but not on the commit prior. Points to secp256k1 ar-lib auxillary file:

```
=== configuring in src/secp256k1 (/Users/dunxen/Code/bitcoin/bitcoin/src/secp256k1)
configure: running /bin/sh ./configure --disable-option-checking '--prefix=/usr/local'  '--with-gui=yes' '--with-sqlite=yes' '--without-bdb' '--disable-shared' '--with-pic' '--enable-benchmark=no' '--enable-module-recovery' '--enable-module-schnorrsig' --cache-file=/dev/null --srcdir=.
configure: error: cannot find required auxiliary files: ar-lib
configure: error: ./configure failed for src/secp256k1
```

Environment:

macOS 12.3.1 (21E258) (Intel)

Should I open a separate issue, or does this look just related to my environment?

**Comment by jonatack** [comment]
> EDIT: fanquake suggested `git clean -fx` and then rerunning autogen and configure. That worked!

Cool. Was going to report that the build is clean for me on Debian. Sometimes it's a good idea to make clean or make distclean as a first step when running into issues like this.

**Comment by real-or-random** [comment]
> > EDIT: fanquake suggested `git clean -fx` and then rerunning autogen and configure. That worked!
> 
> Cool. Was going to report that the build is clean for me on Debian. Sometimes it's a good idea to make clean or make distclean as a first step when running into issues like this.

In this case, most probably rerunning `./autogen.sh` was the right fix. It will create `ar-lib` and this is indeed new in this revision.

**Comment by jonatack** [comment]
Good point. My bash alias for fresh rebuilding does autogen && configure <options> && make clean && make.

**Comment by dunxen** [comment]
> In this case, most probably rerunning `./autogen.sh` was the right fix. It will create `ar-lib` and this is indeed new in this revision.

Yeah, make clean was not sufficient in this case (but is my normal goto for these types errors admittedly 😅) 
Definitely `./autogen.sh` saved the day!

**Pull request merged**